codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/smoketests/SmokeTestsRunnerGenerator.kt

```diff
@@ -0,0 +1,174 @@
+            }
+            val expectingSpecificError = testCase.expectation.failure.getOrNull()?.errorId?.getOrNull() != null
+            write("interceptors.add(#T($expectingSpecificError))", RuntimeTypes.HttpClient.Interceptors.SmokeTestsInterceptor)
+        }
+        checkVendorParamsAreCorrect(testCase)
+    }
+
+    /**
+     * Smithy IDL doesn't check that vendor params are found in vendor params shape so we have to check here.
+     */
+    private fun checkVendorParamsAreCorrect(testCase: SmokeTestCase) {
+        if (testCase.vendorParamsShape.isPresent && testCase.vendorParams.isPresent) {
+            val vendorParamsShape = model.getShape(testCase.vendorParamsShape.get()).get()
+            val validVendorParams = vendorParamsShape.members().map { it.memberName }
+            val vendorParams = testCase.vendorParams.get().members.map { it.key.value }
+
+            vendorParams.forEach { vendorParam ->
+                check(validVendorParams.contains(vendorParam)) {
+                    "Smithy smoke test \"${testCase.id}\" contains invalid vendor param \"$vendorParam\", it was not found in vendor params shape \"${testCase.vendorParamsShape}\""
+                }
+            }
+        }
+    }
+
+    private fun renderOperation(operation: OperationShape, testCase: SmokeTestCase) {
+        val operationSymbol = symbolProvider.toSymbol(model.getShape(operation.input.get()).get())
+
+        writer.addImport(operationSymbol)
+        writer.withBlock(".use { client ->", "}") {
+            withBlock("client.#L(", ")", operation.defaultName()) {
+                withBlock("#L {", "}", operationSymbol.name) {
+                    testCase.params.get().members.forEach { member ->
+                        write("#L = #L", member.key.value.toCamelCase(), member.value.render())
+                    }
+                }
+            }
+        }
+    }
+
+    private fun renderCatchBlock(testCase: SmokeTestCase) {
+        val successCriterion = RuntimeTypes.HttpClient.Interceptors.SmokeTestsSuccessException
+        val failureCriterion = getFailureCriterion(testCase)
+        val expected = if (testCase.expectation.isFailure) {
+            failureCriterion
+        } else {
+            successCriterion
+        }
+
+        writer.write("val success = e is #L", expected)
+        writer.write("val status = if (success) \"ok\" else \"not ok\"")
+        printTestResult(
+            sdkId.filter { !it.isWhitespace() },
+            testCase.id,
+            testCase.expectation.isFailure,
+            writer,
+        )
+        writer.write("if (!success) exitCode = 1")
+    }
+
+    /**
+     * Tries to get the specific exception required in the failure criterion of a test.
+     * If no specific exception is required we default to the generic smoke tests failure exception.
+     *
+     * Some smoke tests model exceptions not found in the model, in that case we default to the generic smoke tests
+     * failure exception.
```

Comment:
I think these should be considered invalid models and added to the deny list until fixed